### ISSUE TITLE
JSoup encoder

### ISSUE THREAD

**Issue by ScalaWilliam** [http4s/http4s#5007]
Hi, I wanted to check if there's interest in bringing in a [JSoup](https://jsoup.org/) Document encoder to this project? Happy to make a PR if so.

JSoup has its own `Document` type which is fairly similar to a DOM Document. Typical use case is to take an HTML template, replace some fields, and serve that document.

The code is here and used in Scala Algorithms, alongside the scalatags encoder. https://github.com/ScalaWilliam/rad4s/blob/master/http4s-jsoup-encoder/src/main/scala/com/scalawilliam/rad4s/http4s/JSoupDocumentEncoder.scala

**Comment by rossabaker** [comment]
I've used the library several years ago, and I'd be happy to host it in the org.

The plan is to spin integrations into their own repos for http4s-1.0, and I'm sure that would be the long-term plan for this.  And it's going to be up to the community to maintain some of those integrations: it burns out the core team maintaining modules they don't use.  This polyrepo model is already working in various backends like Finagle, Netty, Armeria, and the jdk-http-client, and it's the future for our codecs.  So if we add it, it'll be up to you to keep it fresh, or find someone who will.

In the short term, it's a Java dependency with a long track record of binary compatibility (at least the version numbers say so).  I've discouraged adding some other integrations to the core repo, but this one is very low risk.  I'm open either to adding it here in series/0.22 and making it part of The Great Schism like everything else, or if you want to practice maintaining a satellite repo, setting that up.
